### ISSUE TITLE
How To Correct for a "Phantom" review

### ISSUE THREAD

**Issue by robmcclel** [boonebgorges/bp-group-reviews#29]
Boone,

We recently had a situation occur where a "phantom" zero-star review made it's way into the system.  I'm assuming that when the review selection in "settings" was toggled, it somehow created a baseline review.  Not sure how it happened, really, so that's just a theory.

The result is we have a three reviews ( a zero and two 5-star reviews) in the database in the group-meta table.  I've found them, but am unsure of what to change.  Here are the table entries:

posted_review:  a:3:{i:0;s:0:"";i:1;i:8;i:2;i:5;}
bpgr_ratings:  a:2:{i:23313;i:5;i:23323;i:5;}
bpgr_how_many_ratings:  3
bpgr_rating:  3.333

How do I identify and correct for the "phantom" zero rating?

Thanks

**Comment by boonebgorges** [comment]
hi @robmcclel  - I don't have a quick answer for this, but here are some pointers:

`groups_get_groupmeta( $group_id, 'posted_review' )` will get you an array of user IDs corresponding to users who have left reviews on the group. It looks like this array has an invalid entry at index 0, so you'll want to pull it up, remove it, and resave. (Do this and other operations in PHP to avoid juggling serialized strings.)

'bpgr_ratings' looks correct, and probably doesn't need to be fixed.

`bpgr_how_many_ratings` and `bpgr_rating` will need to be recalculated. Since you're dealing with a small number of items, I'd just do it manually.

I think that this will probably be enough to fix it.

**Comment by robmcclel** [comment]
Thanks, Boone.

I take it this is the erroneous entry :  i:0;s:0:"";

If I call up the record in phpmyadmin, can I remove that, correct the a:3 to a:2 on that line, and then recalc the other sections and save the record?  Seems like that's the easiest way.  I just didn't want to do that and make it worse... ;)

**Comment by boonebgorges** [comment]
Yes, you can try that, though I personally would far rather do it in PHP. Just be sure to take a backup before making any changes.
